terraform/deployments/govuk-publishing-infrastructure/govuk_mirror_sync.tf

```diff
@@ -0,0 +1,54 @@
+module "govuk_mirror_sync_iam_role" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-eks-role"
+  version = "~> 5.28"
+
+  role_name        = "govuk-mirror-sync"
+  role_description = "Role for govuk-mirror-sync to access S3. Corresponds to govuk-mirror-sync k8s ServiceAccount."
+
+  cluster_service_accounts = {
+    "${local.cluster_name}" = ["apps:govuk-mirror-sync"]
+  }
+
+  role_policy_arns = {
+    govuk_mirror_sync_policy = aws_iam_policy.govuk_mirror_sync.arn
+  }
+}
+
+resource "aws_iam_policy" "govuk_mirror_sync" {
+  name        = "govuk-mirror-sync"
+  description = "Allows mirror sync process to access S3."
+
+  policy = jsonencode({
```

Comment:
optional: you can get TF to do somewhat more ahead-of-time checking if you represent the IAM policy using a [data.aws_iam_policy_document](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) and referring to the its `.json` attribute here. (Also makes the syntax for the policy slightly nicer cos it lets you omit Version and Effect.)

Reply:
Ah yes - I always forget to do this!

Reply:
Oh no worries! Took me quite a while to figure out that that seems to be the safest way to do it. I should go have another look if there's a linter/fixer for it — glad you've reminded me :D

Reply:
Yeah that'd be ace!